### ISSUE TITLE
`Unhandled Promise Rejection Error` when importing `Rudder-SDK-React-Native` in expo development build

### ISSUE THREAD

**Issue by 1abhishekpandey** [expo/expo#21302]
### Summary

Expected behaviour:

- This error, `Unhandled promise rejection [RangeError: Maximum call stack size exceeded.]`, shouldn't be occurring during each app startup in the expo development build.

This is what actually happens:
- When app is built using `expo` and following dependency is imported `import rc from "@rudderstack/rudder-sdk-react-native";`, in the App.js file.
- Each time I run the android app in the development build, an error toast with the message `Unhandled promise rejection [RangeError: Maximum call stack size exceeded.]` appears on the Android emulator.
- When I import the same dependency, on the project built with `npx react-native init` or through `nx` then no such error is displayed.
- Additionally, the error does not occur in the `production` build of the expo app.

### What platform(s) does this occur on?

Android

### SDK Version

47.0.13

### Environment

expo-env-info 1.0.5 environment info:
    System:
      OS: macOS 13.2.1
      Shell: 5.8.1 - /bin/zsh
    Binaries:
      Node: 16.15.1 - ~/.nvm/versions/node/v16.15.1/bin/node
      Yarn: 1.22.19 - /opt/homebrew/bin/yarn
      npm: 9.4.0 - /opt/homebrew/bin/npm
      Watchman: 2023.02.13.00 - /opt/homebrew/bin/watchman
    Managers:
      CocoaPods: 1.11.3 - /opt/homebrew/bin/pod
    SDKs:
      iOS SDK:
        Platforms: DriverKit 22.2, iOS 16.2, macOS 13.1, tvOS 16.1, watchOS 9.1
    IDEs:
      Android Studio: 2022.1 AI-221.6008.13.2211.9514443
      Xcode: 14.2/14C18 - /usr/bin/xcodebuild
    npmPackages:
      expo: ~47.0.12 => 47.0.13 
      react: 18.1.0 => 18.1.0 
      react-native: 0.70.5 => 0.70.5 
    npmGlobalPackages:
      eas-cli: 3.6.0
      expo-cli: 6.3.0
    Expo Workflow: bare

### Minimal reproducible example

1. Create a React-Native app using: `expo init DebugExpoApp`
2. Choose a minimal template: `minimal             bare and minimal, just the essentials to get you started`
3. Install dependency: `npm i @rudderstack/rudder-sdk-react-native`
4. Open App.js file (located at the root) and add following import and save it: `import rc from "@rudderstack/rudder-sdk-react-native";`
5. Run it on Android emulator: `npm run android`
6. Observe the error both in the Android emulator and console: `ERROR  Unhandled promise rejection [RangeError: Maximum call stack size exceeded.]`
<img width="395" alt="image" src="https://user-images.githubusercontent.com/64667840/220081305-056f904e-c49d-40da-bfee-3bf47507622b.png">

**Comment by 1abhishekpandey** [comment]
We debugged the issue more from our side and when we changed the babel configuration of the npm package `"@rudderstack/rudder-sdk-react-native"`: from `"useBuiltIns": "usage"` to `"useBuiltIns": "entry"`. Then no `Unhandled promise rejection` error is shown.

Here is the `babel.config.json` file of rudder-sdk-react-native (before making any changes):

> {
  "presets": [
    [
      "@nrwl/react/babel",
      {
        "runtime": "automatic",
        "useBuiltIns": "entry"
      }
    ]
  ],
  "plugins": [],
  "env": {
    "test": {
      "presets": ["module:metro-react-native-babel-preset"]
    }
  }
}

While this solution resolves our issue, we are still interested in understanding why this problem occurs exclusively with an Expo-built app in the development environment.
